File: pkg/codegen/testing/utils/testdata/deferred-outputs-pp/nodejs/deferred-outputs.ts
import * as pulumi from "@pulumi/pulumi";
import { First } from "./first";
import { Second } from "./second";

const [secondPasswordLength, resolveSecondPasswordLength] = pulumi.deferredOutput<number>();
const first = new First("first", {passwordLength: secondPasswordLength});
const second = new Second("second", {petName: first.petName});
resolveSecondPasswordLength(second.passwordLength);
const [loopingOverMany, resolveLoopingOverMany] = pulumi.deferredOutput<Array<number>>();
const another = new First("another", {passwordLength: loopingOverMany.apply(loopingOverMany => loopingOverMany.length)});
const many: Second[] = [];
for (const range = {value: 0}; range.value < 10; range.value++) {
    many.push(new Second(`many-${range.value}`, {petName: another.petName}));
}
resolveLoopingOverMany(pulumi.output(many.map((v, k) => [k, v]).map(([_, v]) => (v.passwordLength))));

File: pkg/codegen/testing/utils/testdata/deferred-outputs-pp/go/deferred-outputs.go
package main

import (
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumix"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		secondPasswordLength, resolveSecondPasswordLength := pulumi.DeferredOutput[int](ctx)
		first, err := NewFirst(ctx, "first", &FirstArgs{
			PasswordLength: pulumix.Cast[pulumi.IntOutput](secondPasswordLength),
		})
		if err != nil {
			return err
		}
		second, err := NewSecond(ctx, "second", &SecondArgs{
			PetName: first.PetName,
		})
		if err != nil {
			return err
		}
		resolveSecondPasswordLength(second.PasswordLength)
		loopingOverMany, resolveLoopingOverMany := pulumi.DeferredOutput[[]int](ctx)
		another, err := NewFirst(ctx, "another", &FirstArgs{
			PasswordLength: pulumix.Cast[pulumi.IntOutput](loopingOverMany.ApplyT(func(loopingOverMany []int) (pulumi.Int, error) {
				return pulumi.Int(len(loopingOverMany)), nil
			}).(pulumi.IntOutput)),
		})
		if err != nil {
			return err
		}
		var many []*Second
		for index := 0; index < 10; index++ {
			key0 := index
			_ := index
			__res, err := NewSecond(ctx, fmt.Sprintf("many-%v", key0), &SecondArgs{
				PetName: another.PetName,
			})
			if err != nil {
				return err
			}
			many = append(many, __res)
		}
		resolveLoopingOverMany("TODO: For expression")
		return nil
	})
}
